commands/audit/jas/common.go

```diff
@@ -274,3 +274,30 @@ func GetExcludePatterns(module jfrogappsconfig.Module, scanner *jfrogappsconfig.
 	}
 	return excludePatterns
 }
+
+func SetAnalyticsMetricsDataForAnalyzerManager(technologies []coreutils.Technology) {
+	// For now only one value is supported.
+	if len(technologies) != 1 {
+		return
+	}
+	technology := technologies[0]
+	err := os.Setenv("AM_PACKAGE_MANAGER", technology.String())
+	if err != nil {
+		log.Debug(fmt.Sprintf("failed setting AM_PACKAGE_MANAGER as environment variable. Cause: %s", err.Error()))
+	}
+	err = os.Setenv("AM_LANGUAGE", string(coreutils.TechnologyToLanguage(technology)))
+	if err != nil {
+		log.Debug(fmt.Sprintf("failed setting AM_LANGUAGE as environment variable. Cause: %s", err.Error()))
+	}
+}
+
+func ResetAnalyticsMetricsDataForAnalyzerManager() {
+	err := os.Setenv("AM_PACKAGE_MANAGER", "")
+	if err != nil {
+		log.Debug(fmt.Sprintf("failed setting AM_PACKAGE_MANAGER as environment variable. Cause: %s", err.Error()))
+	}
+	err = os.Setenv("AM_LANGUAGE", "")
+	if err != nil {
+		log.Debug(fmt.Sprintf("failed setting AM_LANGUAGE as environment variable. Cause: %s", err.Error()))
+	}
+}
```

Comment:
```suggestion
func SetAnalyticsMetricsDataForAnalyzerManager(technologies []coreutils.Technology) func() {
	if len(technologies) != 1 {
        // Only report analytics for one technology at a time.
		return
	}
	technology := technologies[0]
	resetAnalyzerManagerPackageManagerVar := SetEnvWithResetCallback(jfPacakgeManagerEnvVariable, technology.String())
	resetAnalyzerManagerLanguageVar := SetEnvWithResetCallback(jfLanguageEnvVariable, technology.Language)
	return func() {
		resetAnalyzerManagerPackageManagerVar()
		resetAnalyzerManagerLanguageVar()
	}
}
```

* create const variables at `analyzermanager.go` and use it
* You can return a func when setting the variables that will reset them
* add general `SetEnvWithResetCallback` to `utils.go` and use it:
```
func SetEnvWithResetCallback(key, value string) func() {
	oldValue, exist := os.LookupEnv(key)
    errMsg := "failed %s %s as environment variable. Cause: %s"

    if err := os.Setenv(key, value); err != nil {
        log.Debug(fmt.Sprintf(errMsg, "setting", key, err.Error()))
        return
    }

	if exist {
		return func() {
            if err := os.Setenv(key, oldValue); err != nil {
                log.Debug(fmt.Sprintf(errMsg, "setting", key, err.Error()))
            }
		}
	}

	return func() {
		if err := os.Unsetenv(key); err != nil {
            log.Debug(fmt.Sprintf(errMsg, "unsetting", key, err.Error()))
        }
	}
}
```